lib/auth/v4/queryAuthCheck.js

```diff
@@ -23,6 +24,13 @@ function check(request, log, data) {
     if (Object.keys(authParams).length !== 5) {
         return { err: errors.InvalidArgument };
     }
+
+    const token = data['X-Amz-Security-Token'];
```

Comment:
all the actual params are case-sensitive, nothing in AWS says its case-insensitive, from that, since we do not have any issue identified for that, i assume all the tools follow this rule :)